hardware/setup/snap_build.tcl

```diff
@@ -16,13 +16,15 @@
 #
 #-----------------------------------------------------------
 
-set log_dir      $::env(LOGS_DIR)
-set log_file     $log_dir/snap_build.log
-set fpgacard     $::env(FPGACARD)
-set sdram_used   $::env(SDRAM_USED)
-set nvme_used    $::env(NVME_USED)
-set bram_used    $::env(BRAM_USED)
-set ila_debug    [string toupper $::env(ILA_DEBUG)]
+set root_dir      $::env(SNAP_HARDWARE_ROOT)
+set log_dir       $::env(LOGS_DIR)
+set log_file      $log_dir/snap_build.log
+set fpgacard      $::env(FPGACARD)
+set sdram_used    $::env(SDRAM_USED)
+set nvme_used     $::env(NVME_USED)
+set bram_used     $::env(BRAM_USED)
+set factory_image [string toupper $::env(FACTORY_IMAGE)]
```

Comment:
Shouldn't we add a check to the Makefile, that FACTORY_IMAGE exists? (Like we do that for other environment variables that need to exist). Otherwise we should add a check for existance here.